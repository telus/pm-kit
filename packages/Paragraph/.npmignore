*.stories.{js,mdx}
rollup.config.js
Paragraph.jsx
__tests__/