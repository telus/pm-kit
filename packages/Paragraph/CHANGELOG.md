# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 1.0.0 (2020-01-16)


### Features

* **paragraph:** added Paragraph package ([#25](https://github.com/telus/pm-kit/issues/25)) ([a61c76a](https://github.com/telus/pm-kit/commit/a61c76ab02502b4de380459a17a5326fcea570b1))
