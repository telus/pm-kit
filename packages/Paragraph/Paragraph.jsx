import React, { forwardRef } from 'react'
import PropTypes from 'prop-types'
import { css } from '@emotion/core'
import { size as fontSize, weight as fontWeight } from '@pm-kit/typography'
import * as colors from '@pm-kit/colours'

const Paragraph = ({ children, position, size, weight, color, decoration, forwardedRef, ...rest }) => {
  const paragraph = css`
    text-align: ${position};
    width: 100%;
    text-decoration: ${decoration};
    font-size: ${size};
    font-weight: ${weight};
    color: ${color};
  `

  return (
    <p css={paragraph} {...rest}>
      {children}
    </p>
  )
}

Paragraph.propTypes = {
  /**
   * The position of text
   */
  position: PropTypes.oneOf(['left', 'center', 'right', 'justify']),
  color: PropTypes.string,
  /**
   * The decoration of text. Example: add an underline below text.
   */
  decoration: PropTypes.string,
  /**
   * The font-weight of text. Example: bold.
   */

  weight: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /**
   * The text size.
   */
  size: PropTypes.string,
  /**
   * The text.
   */
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
}

Paragraph.defaultProps = {
  position: 'left',
  color: colors.parkGreen,
  decoration: 'none',
  weight: fontWeight.normal,
  size: fontSize.bodyMedium,
  children: '',
}

const ParagraphWithRef = forwardRef((props, ref) => <Paragraph {...props} forwardedRef={ref} />)

export default ParagraphWithRef
