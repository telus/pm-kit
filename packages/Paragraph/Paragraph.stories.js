import React from 'react'
import { withKnobs, text, select } from '@storybook/addon-knobs'
import Paragraph from './Paragraph.jsx'
import * as colors from '../colours/colours'
import { size, weight } from '../typography/typography'
export default {
  title: 'Design System|Paragraph',
  component: Paragraph,
  decorators: [withKnobs],
}

export const Default = () => <Paragraph>Test</Paragraph>

export const Playground = () => {
  return (
    <Paragraph
      children={text('Label', '')}
      position={select('Position', ['left', 'center', 'right', 'justify'], 'left')}
      size={select(
        'Size',
        [size.bodySmall, size.bodyMedium, size.bodyLarge, size.h1, size.h2, size.h3, size.h4],
        size.bodySmall
      )}
      color={select(
        'colors',
        [
          colors.greyBlue,
          colors.parkGreen,
          colors.lilyGreen,
          colors.softSandBrown,
          colors.lightTan,
          colors.offWhite,
          colors.white,
          colors.red,
        ],
        colors.parkGreen
      )}
      decoration={select('decoration', ['underline', 'none'], 'none')}
      weight={select('Weight', [weight.normal, weight.bold], weight.normal)}
    />
  )
}

Playground.story = {
  name: 'playground',
  parameters: { docs: { page: null, disable: true } },
}
