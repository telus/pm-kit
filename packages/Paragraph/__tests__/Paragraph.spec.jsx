import React from 'react'
import { render } from 'enzyme'

import Paragraph from '../Paragraph'

describe('Paragraph', () => {
  it('renders', () => {
    const paragraph = render(<Paragraph />)

    expect(paragraph).toMatchSnapshot()
  })
})
