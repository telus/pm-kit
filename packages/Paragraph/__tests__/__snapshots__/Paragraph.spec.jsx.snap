// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph renders 1`] = `
.emotion-0 {
  text-align: left;
  width: 100%;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.125rem;
  font-weight: 300;
  color: #034045;
}

<p
  class="emotion-0"
/>
`;
